File: backend/app/app/schemas/event.py
from datetime import datetime, time
from uuid import UUID
from pydantic import BaseModel


class EventBase(BaseModel):
    event_path: str
    video_title: str | None = None
    rtsp_link: str | None = None
    rtsp_time: datetime | None = None
    video_time: time | None = None

    class Config:
        orm_mode = True


class EventUpdate(EventBase):
    pass


class Event(EventBase):
    id: UUID
    task_id: UUID


class EventCreate(Event):
    pass

File: frontend/src/types/Event.ts
export interface IEventBase {
    event_path: string
    description?: string
    video_title?: string
    rtsp_link?: string
    rtsp_time?: string
    video_time?: string
}

export interface IEventUpdate extends IEventBase {

}

export interface IEvent extends IEventBase {
    id: string
    task_id: string
}

export interface IEventCreate extends IEvent {

}